src/main/java/edu/ie3/datamodel/io/source/csv/CsvWeatherSource.java

```diff
@@ -217,9 +218,12 @@ private Try<Stream<Map<String, String>>, SourceException> buildStreamWithFieldsT
               allRows, timeCoordinateIdExtractor, entityClass.getSimpleName(), "UUID")
           .map(Set::parallelStream);
     } catch (IOException e) {
-      log.warn(
-          "Cannot read file to build entity '{}': {}", entityClass.getSimpleName(), e.getMessage());
-      return Success.of(Stream.empty());
+      return Failure.of(
+          new SourceException(
+              "Cannot read file to build entity '"
+                  + entityClass.getSimpleName()
+                  + "': "
+                  + e.getMessage()));
```

Comment:
Please hand over the causal exception with the constructor as `new SourceException("...", e)`